huaweicloud/services/ram/resource_huaweicloud_ram_resource_share.go

```diff
@@ -459,20 +479,40 @@ func resourceRAMShareDelete(ctx context.Context, d *schema.ResourceData, meta in
 	return nil
 }
 
-func disassociateRAMShare(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
-	disassociateRAMShareHttpUrl := "v1/resource-shares/{resource_share_id}/disassociate"
-	disassociateRAMSharePath := client.Endpoint + disassociateRAMShareHttpUrl
-	disassociateRAMSharePath = strings.ReplaceAll(disassociateRAMSharePath, "{resource_share_id}", d.Id())
-	disassociateRAMShareOpt := golangsdk.RequestOpts{
+func associatePrincipalsAndResourceUrns(client *golangsdk.ServiceClient, resourceId string, principals,
+	resourceUrns []interface{}) error {
+	associateHttpUrl := "v1/resource-shares/{resource_share_id}/associate"
+	associatePath := client.Endpoint + associateHttpUrl
+	associatePath = strings.ReplaceAll(associatePath, "{resource_share_id}", resourceId)
+	associateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: map[string]interface{}{
+			"principals":    principals,
+			"resource_urns": resourceUrns,
```

Comment:
Does it works well when principals or resourceUrns is nil or an empty slice?

Reply:
It works.
I change the test function to update `principals` and `resource_urns` single